src/test/java/plugins/ConfigFileProviderTest.java

```diff
@@ -91,4 +94,36 @@ private String createPipelineAndGetConsole(final MavenSettingsConfig mvnConfig)
         return b.getConsole();
     }
 
+    @Test
+    public void testCustomConfigFile() {
+        final CustomConfig customConfig = this.createCustomConfig();
+        final String jobLog = this.createPipelineAndGetConsole(customConfig);
+
+        assertThat(jobLog, containsString(CUSTOM_CONF_CONTENT));
```

Comment:
I presume you intend to verify the content of the file is injected correctly but the reason this passes is the filename is the same as the content  that is echoed with the command.